CRM/Core/Form/Task/PDFLetterCommon.php

```diff
@@ -198,7 +198,17 @@ public static function setDefaultValues() {
    */
   public static function formRule($fields, $files, $self) {
     $errors = [];
-    $template = CRM_Core_Smarty::singleton();
+    $deprecatedTokens = [
+      '{case.status_id}' => '{case.status_id:label}',
+      '{case.case_type_id}' => '{case.case_type_id:label}',
+    ];
+    foreach ($deprecatedTokens as $token => $replacement) {
+      if (strpos($fields['html_message'], $token) !== FALSE) {
+        $errors['html_message'] = ts('Replaced token usage - '
+        . '{case.status_id} replaced by {case.status_id:label}<br>'
+        . '{case.case_type_id} replaced by {case.case_type_id:label}');
```

Comment:
This formatting with separate lines with `.` doesn't work with civistrings.

And not sure about the wording. Maybe "Token {case.status_id} is no longer supported - use {case.status_id:label} instead."

Also minor thing but this loop will always run twice and even if only one was replaced it will say both were.

And I'm guessing you're aware but just noting this doesn't prevent using them when creating message templates or in case emails.

Reply:
@demeritcowboy yeah I was gonna follow up with case emails once I'd worked through the pdf side of things & on the loop / both thing - I was kinda figuring this was edge case so it wouldn't matter much. I'll fix the wording & . as you say though

Reply:
@demeritcowboy on second thoughts I did the email action as well - here is how it looks now.



![image](https://user-images.githubusercontent.com/336308/132427459-62cdb453-2ca2-4001-a68c-06e0fb3b7331.png)